sopht_mpi/simulator/immersed_body/rigid_body/rigid_body_forcing_grids.py

```diff
@@ -123,3 +120,161 @@ def get_maximum_lagrangian_grid_spacing(self):
         """Get the maximum Lagrangian grid spacing"""
         # ds = radius * dtheta
         return self.cylinder.radius * (2.0 * np.pi / self.num_lag_nodes)
+
+
+SupportedRigidBody3D = Union[ea.Cylinder, ea.Sphere]
+
+
+class ThreeDimensionalRigidBodyForcingGrid(ImmersedBodyForcingGrid):
+    """Class for forcing grid of a 3D rigid body with cross-section."""
+
+    # Will be set in derived classes
+    # num_lag_nodes: int = NotImplementedError
+
+    def __init__(self, grid_dim: int, rigid_body: SupportedRigidBody3D) -> None:
+        if grid_dim != 3:
+            raise ValueError(
+                "Invalid grid dimensions. 3D Rigid body forcing grid is only "
+                "defined for grid_dim=3"
+            )
+        self.rigid_body = rigid_body
+        super().__init__(grid_dim)
+        self.local_frame_relative_position_field = np.zeros_like(self.position_field)
+        self.global_frame_relative_position_field = np.zeros_like(self.position_field)
+
+    def compute_lag_grid_position_field(self) -> None:
+        """Computes location of forcing grid for the rigid body boundary"""
+        self.global_frame_relative_position_field[...] = np.dot(
+            self.rigid_body.director_collection[:, :, 0].T,
+            self.local_frame_relative_position_field,
+        )
+        self.position_field[...] = (
+            self.rigid_body.position_collection
+            + self.global_frame_relative_position_field
+        )
+
```

Comment:
How is the code for this grid and the sphere grid different from what exists in sopht? I have the same comment for other forcing grids even the 2D ones, cant you just import them from sopht?